### PR TITLE
Feature/osx_support

### PR DESCRIPTION
**Pull request by netanelc305**
Add initial OSX support

### CODE REVIEW

**Comment by tklengyel** [comment]
Thanks for the PR. I quickly flipped through and it looks like you have a lot of unrelated changes all throughout the codebase. Please don't do that as it makes reviewing the PR very hard. Also, please make fix the issues spotted by the CI.

**Comment by netanelc305** [comment]
> Thanks for the PR. I quickly flipped through and it looks like you have a lot of unrelated changes all throughout the codebase. Please don't do that as it makes reviewing the PR very hard. Also, please make fix the issues spotted by the CI.

Yeah sorry about that, my IDE decided to reformat some files. It should be fine now, i fixed all error and tested locally.

File: libvmi/os/osx/memory.c [hunk 1]
```diff
@@ -0,0 +1,147 @@
+/* The LibVMI Library is an introspection library that simplifies access to
+ * memory in a target virtual machine or in a file containing a dump of
+ * a system's physical memory.  LibVMI is based on the XenAccess Library.
+ *
+ * Copyright 2011 Sandia Corporation. Under the terms of Contract
```
Comment (tklengyel) [inline libvmi/os/osx/memory.c:5]: You don't need to add this Sandia Copyright to new files unless you copied portions of it from other files within the project.

**Comment by tklengyel** [comment]
Thanks, looks good. Merging.

**Pull request merged**